status/versions.go

```diff
@@ -0,0 +1,116 @@
+package status
+
+import (
+	"encoding/json"
+	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/kubernetes"
+	"io/ioutil"
+	"k8s.io/api/core/v1"
+	kube "k8s.io/client-go/kubernetes"
+	"net/http"
+	"regexp"
+)
+
+type serviceProduct func() (*ProductInfo, error)
+
+func getVersions() {
+	products := []serviceProduct{
+		istioVersion,
+		prometheusVersion,
+		kubernetesVersion,
+	}
+	for _, prod := range products {
+		getVersionProduct(prod)
+	}
+}
+
+func getVersionProduct(serviceProduct serviceProduct) {
+	productInfo, err := serviceProduct()
+	if err == nil {
+		info.Products = append(info.Products, *productInfo)
+	}
+}
+
+func istioVersion() (*ProductInfo, error) {
+	product := ProductInfo{}
+	istioConfig := config.Get().Products.Istio
+	serviceInfo, err := getService(istioConfig.ServiceNamespace, istioConfig.ServiceVersion)
+	if err == nil {
+		resp, err := http.Get("http://" + serviceInfo.ClusterIP + ":9093/version")
+		if err == nil {
+			defer resp.Body.Close()
+			body, err := ioutil.ReadAll(resp.Body)
+			if err == nil {
+				product.Name = "Istio"
+				product.FullVersion = string(body)
+				re := regexp.MustCompile("[0-9]\\.[0-9]\\.[0-9]")
+				product.Version = re.FindStringSubmatch(string(body))[0]
```

Comment:
Maybe some defensive code here on regexp matching , in the case the version format changes suddenly (which would default to 'unknown' then)

Reply:
Added condition @jotak